src/device.rs

```diff
@@ -1017,6 +1017,65 @@ impl<'a, E: StreamSample> TxStream<'a, E> {
         }
     }
 
+    /// Write a burst of samples whose transmission 
+    /// will begin at the given device timestamp
+    ///
+    /// It is not necessary to activate the stream beforehand.  
+    /// This call will block if the device buffers are full.  
+    /// `at_ns` a nanosecond precision device timestamp at which 
+    /// the device is to begin the transmission.  
+    /// `buffers` contains one source slice for each channel of the stream.
+    ///
+    /// Returns the number of samples written, which may be smaller than the size of the passed arrays.
+    /// 
+    /// # Notes
+    /// Internally SoapySDR will do the following:
+    ///  * Activate a tx stream.
+    ///  * Write the samples in the given buffers.
+    ///  * Deactivate the stream.
+    ///
+    /// An inital timestamp may be obtained using 
+    /// [get_hardware_time](Device::get_hardware_time)
+    ///
+    /// # Panics
+    ///  * If `buffers` is not the same length as the `channels` array passed to `Device::rx_stream`.
+    ///  * If all the buffers in `buffers` are not the same length.
+    
+    pub fn write_burst(
+        &mut self,
+        buffers: &[&[E]],
+        at_ns: i64,
+        timeout_us: i64,
+    ) -> Result<usize, Error> {
+        assert!(
+            buffers.len() == self.nchannels,
+            "Number of buffers must equal number of channels on stream"
+        );
+
+        // configure the flags for burst mode
+        self.flags = SOAPY_SDR_TX as i32 + SOAPY_SDR_HAS_TIME as i32 + SOAPY_SDR_END_BURST as i32;
```

Comment:
Regarding SOAPY_SDR_TX, I  chucked it on there for good measure not for any deeper reason  :-).
I could add a type alias or something that indicates what they are to be used for.

Yeah the whole flag situation is rather untidy...

From what i can tell SoapyUHD resets the flags after each call to [writeStream](https://github.com/pothosware/SoapyUHD/blob/master/SoapyUHDDevice.cpp) and readStream, so that would not be that much of a issue, but other wrappers might not do it that way...  
The only place the flags are actually used to convey anything (at least in SoapyUHD) is readStreamStatus (exclusive to TX streams and looks rather costly), where they are used to retrieve Error codes from async metadata (i.e. burst late etc).
If this applies to all/most wrappers we might be fine just parsing them as additional error codes if they are something other than 0 after the call.